plugins/repository-gcs/src/main/java/io/crate/gcs/GCSRepository.java

```diff
@@ -0,0 +1,138 @@
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.gcs;
+
+import java.net.URI;
+import org.elasticsearch.cluster.metadata.RepositoryMetadata;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.blobstore.BlobPath;
+import org.elasticsearch.common.blobstore.BlobStore;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Setting.Property;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.indices.recovery.RecoverySettings;
+import org.elasticsearch.repositories.blobstore.BlobStoreRepository;
+
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+
+/**
+ * Google Cloud Storage implementation of the BlobStoreRepository
+ * <p>
+ * This repository supports the following settings
+ * <dl>
+ * <dt>{@code bucket}</dt><dd>Bucket name</dd>
+ * <dt>{@code base_path}</dt><dd>Base path (blob name prefix) in the bucket</dd>
+ * <dt>{@code private_key_id}</dt><dd>Private key in PKCS 8 format from the google service account credentials.</dd>
+ * <dt>{@code private_key}</dt><dd>Private key from the google service account credentials.</dd>
+ * <dt>{@code client_id}</dt><dd>client id from the google service account credentials.</dd>
+ * <dt>{@code client_email}</dt><dd>client email from the google service account credentials.</dd>
+ * <dt>{@code token_uri}</dt><dd>Endpoint oauth token URI, only used for testing to connect to an alternative oauth provider.</dd>
+ * <dt>{@code endpoint}</dt><dd>Endpoint root URL, only used for testing to connect to an alternative storage.</dd>
+ * </dl>
+ */
+public class GCSRepository extends BlobStoreRepository {
+
+    static final Setting<String> BUCKET_SETTING =
+        Setting.simpleString("bucket", Property.NodeScope);
```

Comment:
minor: I think none of these settings should have a `NodeScope`.

Reply:
👍

Reply:
That was taken over from the OpenSearch settings https://github.com/opensearch-project/OpenSearch/blob/0ee69c6a7e1f293a2389ee24a7f3971fd6a52afe/plugins/repository-gcs/src/main/java/org/opensearch/repositories/gcs/GoogleCloudStorageClientSettings.java#L78

What would be the appropriate scope?

Reply:
We do not have any matching scope for settings related/provided by a statement afaik. So I'd just leave the scope out.

Reply:
I see, make sense.